ga4gh/cli.py

```diff
@@ -816,3 +820,72 @@ def client_main(parser=None):
     else:
         runner = args.runner(args)
         runner.run()
+
+##############################################################################
+# Configuration testing
+##############################################################################
+
+
+class TestConfigExists(unittest.TestCase):
+    """This class implements configuration checks for the GA4GH server. It is
+    structured as a set of unit tests, and relies on the unittest module.
+    However, as some of the tests require state injection (through command
+    line arguments) we need to break the usual unittest pattern and instead
+    implement our own simple test runner.
+    """
+
+    def __init__(self, parser=None):
+        self.parser = parser
+
+    def setUp(self):
+        self.app = flask.Flask(__name__)
+        parser = self.parser if self.parser is not None \
+            else argparse.ArgumentParser(
+                description="GA4GH server configuration validator")
+        parser.add_argument(
+            "--config", "-c", default='DevelopmentConfig', type=str,
+            help="The configuration to use")
+        parser.add_argument(
+            "--config-file", "-f", type=str, default=None,
+            help="The configuration file to use")
+        args = parser.parse_args()
+        config_str = 'ga4gh.serverconfig:{0}'.format(args.config)
+        self.app.config.from_object(config_str)
+        if os.environ.get('GA4GH_CONFIGURATION') is not None:
+            self.app.config.from_envvar('GA4GH_CONFIGURATION')
+        if args.config_file is not None:
+            self.app.config.from_pyfile(args.config_file)
+
+    def test_config_exists(self):
+        self.assertIsInstance(self.app.config["REQUEST_VALIDATION"],
+                              types.BooleanType,
+                              'REQUEST_VALIDATION is not a boolean')
+        self.assertIsNotNone(self.app.config["DATA_SOURCE"],
+                             'DATA_SOURCE must be set')
+
+
+def run_test(test_obj, test):
```

Comment:
We adopted camelCase for this project because that's the convention that the GA4GH protocol uses. I would have preferred snake_case, but having mixed conventions is much worse than one you don't like. I'm not sure where flake8 stands at the moment with camelCase identifiers, if it decides to start checking this, we'll need to update our Travis script.